tests/scripts/all.sh

```diff
@@ -1941,6 +1941,32 @@ component_test_psa_crypto_config_accel_cipher () {
     make test
 }
 
+component_test_psa_crypto_config_accel_aead () {
+    msg "test: MBEDTLS_PSA_CRYPTO_CONFIG with accelerated AEAD"
+
+    # Disable ALG_STREAM_CIPHER and ALG_ECB_NO_PADDING to avoid having
+    # partial support for cipher operations in the driver test library.
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_STREAM_CIPHER
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_ECB_NO_PADDING
+
+    loc_accel_list="ALG_GCM ALG_CCM ALG_CHACHA20_POLY1305 KEY_TYPE_AES KEY_TYPE_CHACHA20"
+    loc_accel_flags=$( echo "$loc_accel_list" | sed 's/[^ ]* */-DLIBTESTDRIVER1_MBEDTLS_PSA_ACCEL_&/g' )
+    make -C tests libtestdriver1.a CFLAGS="$ASAN_CFLAGS $loc_accel_flags" LDFLAGS="$ASAN_CFLAGS"
+
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_DRIVERS
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+
+    scripts/config.py unset MBEDTLS_MODE_GCM
+    scripts/config.py unset MBEDTLS_MODE_CCM
+    scripts/config.py unset MBEDTLS_MODE_CHACHAPOLY
+
+    loc_accel_flags="$loc_accel_flags $( echo "$loc_accel_list" | sed 's/[^ ]* */-DMBEDTLS_PSA_ACCEL_&/g' )"
+    make CFLAGS="$ASAN_CFLAGS -Werror -I../tests/include -I../tests -I../../tests -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_TEST_LIBTESTDRIVER1 $loc_accel_flags" LDFLAGS="-ltestdriver1 $ASAN_CFLAGS"
+
```

Comment:
Please add something like:
```c
    # There's a risk of something getting re-enabled via config_psa.h;
    # make sure it did not happen.
    not grep mbedtls_ccm library/ccm.o
    not grep mbedtls_gcm library/gcm.o
    not grep mbedtls_chachapoly library/chachapoly.o
```
Actually this is not passing now, so this component is not actually testing driver-only AEAD. I suspect GCM gets re-enabled due to the following lines in `config_psa.h`:
```c
#if defined(PSA_WANT_ALG_GCM)
#if !defined(MBEDTLS_PSA_ACCEL_ALG_GCM) || \
    defined(PSA_HAVE_SOFT_KEY_TYPE_AES) || \
    defined(PSA_HAVE_SOFT_KEY_TYPE_ARIA) || \
    defined(PSA_HAVE_SOFT_KEY_TYPE_CAMELLIA)
#define MBEDTLS_PSA_BUILTIN_ALG_GCM 1
#define MBEDTLS_GCM_C
#endif
#endif /* PSA_WANT_ALG_GCM */
```
but I did not investigate.

Reply:
Och no, I used wrong names to unset. Should be:

```
    scripts/config.py unset MBEDTLS_GCM_C
    scripts/config.py unset MBEDTLS_CCM_C
    scripts/config.py unset MBEDTLS_CHACHAPOLY_C
```
But even now they are included in the build.

Is this the correct fix:
```
    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_GCM
    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_CCM
    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_CHACHA20_POLY1305
```

Reply:
> Och no, I used wrong names to unset.

Ah good catch, I re-read it twice and couldn't see it. That should definitely help :)

> Is this the correct fix: [`unset PSA_WANT_ALG_xxx`]

No, that can't be the correct fix. The very goal of this task is to get a build where `PSA_WANT_ALG_GCM` is defined but `MBEDTLS_GCM_C` isn't (and similarly for CCM and ChachaPoly). That's the reason we're doing this dance with `libtestdriver1`. (I realise now that the issue description is could have been clearer, it focuses on the details but forgot to clarify the high-level goal.)

If I understand correctly what `config_psa.h` does (the excerpt I quoted), is that if any of AES, ARIA, Camellia, is present in software, then we include software GCM even if `ACCEL_ALG_GCM` is defined. So, perhaps we need to accelerate more things in order to get there: not just `KEY_TYPE_AES` and `KEY_TYPE_CHACHA20` but also `KEY_TYPE_ARIA` and `KEY_TYPE_CAMELLIA`?

Reply:
I got lost while investigating your proposition, but it turned out the problem was in my environment. I started over with the fresh env and now I can confirm that enabling acceleration for other key types (ARIA, CAMELLIA) solves the problem with included default support for `GCM`, `CCM` and  `CHACHAPOLY`.
Build and additional checks are passing and there is a problem only with one test from `test_suite_ssl`:
```
DTLS Handshake with serialization, tls1_2 ......................... FAILED
  mbedtls_ssl_context_save( &(server.ssl), NULL, 0, &context_buf_len ) == MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL
  at step 1, line 2520, test_suite_ssl.c
```

The same problem was encountered here:
https://github.com/Mbed-TLS/mbedtls/pull/6374#issuecomment-1259501690
and as a result we disabled `MBEDTLS_SSL_CONTEXT_SERIALIZATION` for non-AEAD build (https://github.com/Mbed-TLS/mbedtls/pull/6374/commits/d582a01073ce8db8698e28cc92430ca283eb7dcf)
This was ok for that PR because AEAD was disabled, but here AEAD is available, but using drivers.
So the simplest solution would be to add:
`scripts/config.py unset MBEDTLS_SSL_CONTEXT_SERIALIZATION`
and I confirmed that now tests are passing. 
Taking into account that this change will go in https://github.com/Mbed-TLS/mbedtls/pull/6374/commits/d582a01073ce8db8698e28cc92430ca283eb7dcf and we have disabled:
```
    scripts/config.py unset MBEDTLS_GCM_C
    scripts/config.py unset MBEDTLS_CCM_C
    scripts/config.py unset MBEDTLS_CHACHAPOLY_C
```
this approach looks good, but on the other hand AEAD is available by drivers, so maybe the test incorrectly selects MAC cipher suite?

Reply:
I think it's fine to unset `MBEDTLS_SSL_CONTEXT_SERIALIZATION` for now, this can be addressed in a later task in this EPIC, for example I think #6322 covers that.

Let's progress incrementally: first, get a build with driver-only AEAD that works and passes tests, even if it's just a subset of the library that's build, and only a subset of of tests that actually run (currently a lot of tests are skipped because they depend on `GCM_C` even though they could in principle work with the PSA-based GCM that's available) - this is this task. Then, module by module we'll make everything achieve feature and testing parity with software-based AEAD - this is the rest of the EPIC :)